testing/src/main/java/org/oppia/android/testing/AccessibilityTestRule.kt

```diff
@@ -0,0 +1,49 @@
+package org.oppia.android.testing
+
+import android.os.Build
+import androidx.test.espresso.accessibility.AccessibilityChecks
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+/**
+ * JUnit rule to enable [AccessibilityChecks] on Espresso.
+ * Reference: https://developer.android.com/training/testing/espresso/accessibility-checking
+ */
+class AccessibilityTestRule : TestRule {
```

Comment:
Is it possible to have tests for this class? It can be tricky to test a JUnit rule, but really valuable since it's actually easy to mess them up & that has significant repercussions on the entire codebase. Even something as simple as forgetting or conditioning out the 'evaluate()' call will result in all tests passing.

Reply:
@BenHenning I tried searching about how to write test cases for TestRule but I am unable to find any resource. I have gone through some repos which contains `TestRule` mentioned [here](https://medium.com/mobile-app-development-publication/all-about-testrule-a-steroid-before-after-a74ef421e3e5) but in these repo's also there are not test cases for `TestRule`.

Also considering that all the functions are `private` in `AccessibilityTestRule` how do we actually test it?

Do you have any suggestions?

Reply:
We would need to be able to check whether the accessibility checks are enabled at runtime. Can we do that?

If so, the best way would probably be to manually call apply(). It's a bit hacky, but working through the test framework will be a lot harder, I think. That being said, constructing a Statement/Description might not be possible.